vpr/src/route/rr_graph_reader.cpp

```diff
@@ -438,6 +470,20 @@ void process_edges(pugi::xml_node parent, const pugiutil::loc_data & loc_data,
         //set edge in correct rr_node data structure
         device_ctx.rr_nodes[source_node].set_edge_sink_node(num_edges_for_node[source_node], sink_node);
         device_ctx.rr_nodes[source_node].set_edge_switch(num_edges_for_node[source_node], switch_id);
+
+        // Read the metadata for the edge
+        auto metadata = get_single_child(edges, "metadata", loc_data, OPTIONAL);
+        if (metadata) {
+            auto edges_meta = get_first_child(metadata, "meta", loc_data);
+            while (edges_meta) {
+                auto key = get_attribute(edges_meta, "name", loc_data).as_string();
+
+                vpr::add_edge_metadata(source_node, sink_node, switch_id,
```

Comment:
add_rr_edge_metadata would be a better name (we have lots of types of edges in vpr).